File: Ejercicios GO/11.Listas.go
package main

import "fmt"

func main() {
	// Definimos un slice de strings que contiene los días de la semana.
	Lista1 := []string{"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado", "Domingo"}
	fmt.Println(Lista1[4]) // Imprime "Viernes".

	// Definimos otro slice de strings con los mismos días de la semana.
	Lista2 := []string{"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado", "Domingo"}
	fmt.Println(Lista2) // Imprime la lista completa.

	// Definimos un tercer slice de strings y mostramos los primeros tres días.
	Lista3 := []string{"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado", "Domingo"}
	fmt.Println(Lista3[0:3]) // Imprime ["Lunes", "Martes", "Miercoles"].

	// Creamos un slice que puede contener diferentes tipos de datos.
	Lista4 := []interface{}{
		"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado", 1, 2, 3,
		[]interface{}{"Esteban", 0.2, 2.25, true}, // Último elemento es otro slice.
	}

	// Accedemos al décimo elemento de Lista4 y lo convertimos a un slice de interface{}.
	subLista := Lista4[9].([]interface{})
	fmt.Println(subLista[0:3]) // Imprime ["Esteban", 0.2, 2.25].

	// Imprimimos elementos de las listas de forma separada.
	fmt.Println(Lista1[4]) // Imprime "Viernes".
	fmt.Println(Lista2) // Imprime la lista completa.

	// Imprimimos el primer y cuarto elemento de Lista3.
	fmt.Print(Lista3[0], " ") // Imprime "Lunes" seguido de un espacio.
	fmt.Println(Lista3[3]) // Imprime "Jueves".

	// Imprimimos elementos de Lista4 y de la sublista.
	fmt.Print(Lista4[0], " ") // Imprime "Lunes".
	fmt.Print(Lista4[4], " ") // Imprime "Viernes".
	fmt.Print(subLista[0], " ") // Imprime "Esteban".
	fmt.Println(subLista[3]) // Imprime "true".
}

File: Ejercicios C/11.Listas.cpp
#include <iostream>
#include <string>
#include <vector> // Agregar esta librería para usar vectores
using namespace std;

int main() {
    // Definir los arreglos
    string Lista1[] = {"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado"};
    string Lista2[] = {"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado"};
    string Lista3[] = {"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado"};
    
    // Mostrar el valor en la posición 4 de Lista1
    cout << "Elemento en la posición 4 de Lista1: " << Lista1[4] << endl;

    // Mostrar la lista completa (Lista2)
    cout << "Lista2 completa: ";
    for (int i = 0; i < 6; i++) {
        cout << Lista2[i] << " ";
    }
    cout << endl;

    // Mostrar los elementos de la posición 0 y 3 de Lista3
    cout << "Elemento en la posición 0 y 3 de Lista3: " << Lista3[0] << ", " << Lista3[3] << endl;

    // Definir un vector que contiene los días de la semana
    vector<string> Lista4 = {"Lunes", "Martes", "Miercoles", "Jueves", "Viernes", "Sabado"};
    
    // Definir un vector con diferentes detalles en formato de string
    vector<string> Lista4_detalles = {"Esteban", "0.2", "2.25", "true"};
    
    // Mostrar elementos de Lista4
    cout << "Elemento en la posición 0 y 4 de Lista4: " << Lista4[0] << ", " << Lista4[4] << endl;

    // Mostrar detalles de Lista4_detalles
    cout << "Detalles de Lista4: ";
    for (const string& detalle : Lista4_detalles) {
        cout << detalle << " ";
    }
    cout << endl;

    return 0;
}
